service/tracker/execution.go

```diff
@@ -159,7 +171,40 @@ func (e *Execution) processNext() error {
 	// Dump the block execution record into our cache and push all trie updates
 	// into our update queue.
 	e.records[blockID] = record
+	if e.execClient != nil {
+		e.log.Debug().Hex("block_id", blockID[:]).Msg("fetching updates from data sync")
+		e.log.Debug().Int("updates", len(record.TrieUpdates)).Msg("got trie updates from GCP")
+		// get Trie updates from exec data sync
+		req := &access.GetExecutionDataByBlockIDRequest{BlockId: blockID[:]}
+		res, err := e.execClient.GetExecutionDataByBlockID(context.Background(), req)
+		if err != nil {
+			return fmt.Errorf("could not get execution data from access node: %w", err)
+		}
 
+		execTrieUpdates := make(map[string]bool, 0)
+		// collect updates before pushing!
+		for _, chunk := range res.GetBlockExecutionData().ChunkExecutionData {
+			convertedChunk, err := convert.MessageToChunkExecutionData(chunk, e.chain)
+			if err != nil {
+				return fmt.Errorf("unable to convert execution data chunk : %w", err)
+			}
+			if convertedChunk.TrieUpdate != nil {
+				execTrieUpdates[convertedChunk.TrieUpdate.String()] = true
+				e.log.Debug().Str("source", "exec sync").Str("update", convertedChunk.TrieUpdate.String()).Msg("")
+			}
+		}
+		// hash search for matching update
+		for _, gcpUpdate := range record.TrieUpdates {
+			if gcpUpdate != nil && !gcpUpdate.IsEmpty() {
+				e.log.Debug().Str("source", "gcp").Str("update", gcpUpdate.String()).Msg("comparing update")
+				if !execTrieUpdates[gcpUpdate.String()] {
```

Comment:
This comparison isn't complete. We should remove the `gcpUpdate` from the `execTrieUpdates` and see if there any left in `execTrieUpdates` after iterating through record.TrieUpdates, otherwise, if `record.TrieUpdates` is empty, the validation would always pass even if gcpUpdate has some updates.